### ISSUE TITLE
Duplicate field declarations lead to unpredictable behavior.

### ISSUE THREAD

**Issue by michaelstaib** [ChilliCream/hotchocolate#1119]
If a field is declared multiple times then the first declaration wins not the last. In combination with `Include` this can become confusing to the user of the API.

**Comment by michaelstaib** [comment]
#1114

**Comment by drowhunter** [comment]
I have encountered another issue regarding this

Given the following
```
public class UserProfileType : ObjectType<UsperProfile>
{
    protected override void Configure(IObjectTypeDescriptor<UserProfile> descriptor)
    {
       descriptor.Include<UserProfileResolver>();       
    }
}
```
I don't want to pollute my POCO class so I use resolver Types
```
public class UserProfileResolver 
{
        public IQueryable<ValueType<int>> GetFilterNumbers()
        {
	       return new[] { 1, 2, 3, 4, 5, 6, 7, 8, 9 }
              .Select(_ => new ValueType<int>(_))
              .AsQueryable();
        }
}
```
This is fine the `GetFilterNumbers` function appears in the api as expected. Because of the `Include` on the descriptor which includes all methods defined in the Resolver class.

```
query {
  viewer {    
    filterNumbers(order_by:{ value: DESC})  //<!-- does not show up 
    {
      value
    }
  }
}
```

However I would like to use Filtering and Sorting Middleware and maybe add Authorization to it as well. So I modify my Configuring override
```
protected override void Configure(IObjectTypeDescriptor<UserProfile> descriptor)
    {
       descriptor.Include<UserProfileResolver>();    
       descriptor.Field<UserProfileResolver>(f => f.GetFilterNumbers())
            .UseFiltering().UserSorting().Authorize();   
    }
```
This does not work the filter, sorting arguments do not appear in the api and authorize fails to work. Because the include directive at the top.

This is un-intuitive.  Considering although that its nice to be able include all methods from the included resolver class. the only way to attach middleware, or otherwise modify the fields included methods is to add via descriptor override. 

It is assumed that the `descriptor.Field()` methods are referring to the imported fields so that i may further modify them.  But instead it seems that after the import the field definition is written in stone and cannot be modified further.

nothing even warns you about this behavior.

**Comment by drowhunter** [comment]
Removing the Include is undesirable because I would need to explicitly add field overrides for every method even the ones I don't want to modify

**Comment by drowhunter** [comment]
Another scenario I have is that under my UserProfileType all of my fields are protected at the type level using 
`descriptior.Authorize()`

but I have a few fields that require higher level permissions to access

like 
`descriptor.field<ResolverType>(f => f.HighSecurityField).Authorize(Policies.ADMIN);`

this also fails to protect the field for admins

**Comment by michaelstaib** [comment]
Hi @drowhunter this is not really related to this issue. We have another issue open regarding middleware as attribute with a specifically target to allow filtering.

**Comment by drowhunter** [comment]
So your saying that whatever you can do with overrides you need to be able to do with attributes in resolvers.  
I agree.

However.  My issue is more about the consistency.  For example.  The behavior I am asking for above is already how it works in the normal scenario.

```
public class FooType : ObjectType<Foo>
{
    protected override void Configure(IObjectTypeDescriptor<Foo> descriptor)
    {
        descriptor.Field(f => f.Bar()).UseFiltering();
    }
}
```

here i'm accessing a field already defined in the CLR type and modifying it with an override.

**Comment by michaelstaib** [comment]
https://github.com/ChilliCream/hotchocolate/issues/1207

**Comment by drowhunter** [comment]
Thank you.  sorry I seem so aggressive on this one but I really feel like the include function and the way field overrides work currently is strange.

The way it currently works means that having the include statement is not that useful.

Say I had a shared resolver with 20 methods,  but i needed to modify the way "one" functioned (add authorization middle ware for example).  Since the only way to modify them is to NOT use the Include, Without using the Include I would have to manually go to each type that needs the resolver and add in overrides to every shared field just to ensure that they were all actually included were included in my graph.

**Comment by michaelstaib** [comment]
@drowhunter now worries. I think your point is valid. The only concern that I have is the timeline. But, we will try to get a fix for #1207 into the 10 branch if possible. If we however determine it is to big than it will have to go into 11.

**Comment by michaelstaib** [comment]
This one is now fixed with 10.3.0-preview.3
